### PR TITLE
Fix datamining embeds i hope

### LINKED ISSUE #42: Discord datamining embed hitting the character limit crashes the loop
Ideally I'd split it into two, bodged-ly I could just say that there are lots of commits and to look for yourself. Either works. Crashing is not ideal.

### PR DESCRIPTION
**Pull request by Silverdimond**
this **should** fix #42 
and touches on #40

### CODE REVIEW

File: cogs/nickname_handler.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ async def on_member_update(self, before:discord.Member, member:discord.Member):
             if member.nick == new_nickname:
                 return
             try:
-                await member.edit(nick=new_nickname)
+                await member.edit(nick=new_nickname,reason= "Changed by Apple.Py automagicly")
```
Comment (fadedmax) [inline cogs/nickname_handler.py]: Typo

File: cogs/nickname_handler.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ async def fix_user_nickname(self, user:discord.Member, *, force_to_animal:bool=F
 
         # Change their name
         self.logger.info(f"Updating nickname '{current_name}' to '{new_name}' (G{user.guild.id}/U{user.id})")
-        await user.edit(nick=new_name)
+        await user.edit(nick=new_name,reason= "Changed by Apple.Py automagicly")
```
Comment (fadedmax) [inline cogs/nickname_handler.py]: Typo

File: cogs/discord_datamining.py [hunk 3]
```diff
@@ -74,18 +74,30 @@ async def commit_poster_loop(self, embed_title:str, api_url:str, repo_url:str):
...
         with utils.Embed(use_random_colour=True) as embed:
             embed.title = embed_title
-            embed.description = '\n'.join([f"{i['commit']['message']} - [Link]({repo_url}commit/{i['sha']})" for i in new_commits])
+            embed.description = array.pop()
             for sha, body in comment_text:
                 embed.add_field(sha, body, inline=False)
 
         # And send
         channel = self.bot.get_channel(self.VFL_CODING_CHANNEL_ID)
         m = await channel.send(embed=embed)
```
Comment (fadedmax) [inline cogs/discord_datamining.py:97]: Still sending it despite the fact there’s still more to do.

File: cogs/discord_datamining.py [hunk 4]
```diff
@@ -74,18 +74,30 @@ async def commit_poster_loop(self, embed_title:str, api_url:str, repo_url:str):
                 data = await r.json()
             comment_text.extend([(i['commit_id'], re.sub(r"^## (.+)", r"**\1**", i['body'])) for i in data])
         self.logger.info(f"Logging info of {len(comment_text)} comments")
-
-        # Format it all into an embed
+        #get the description
+        pog =  '\n'.join([f"{i['commit']['message']} - [Link]({repo_url}commit/{i['sha']})" for i in new_commits])
```
Comment (fadedmax) [inline cogs/discord_datamining.py]: maybe make a more descriptive variable name

File: cogs/discord_datamining.py [hunk 5]
```diff
@@ -74,18 +74,30 @@ async def commit_poster_loop(self, embed_title:str, api_url:str, repo_url:str):
...
+        if len(pog)>2048:
+            self.logger.info(f"So it turns out our description is longer than 2048 and this isnt really poggers, we need to make it in multiple embeds. It was {len(pog)}")
```
Comment (fadedmax) [inline cogs/discord_datamining.py]: This kinda overbloats it, than again it really doesn’t matter as kae is probably the only ever who will see this lol

**Review by fadedmax — CHANGES_REQUESTED** [review]

File: cogs/discord_datamining.py (hunk 3 above)
Comment (fadedmax) [reply cogs/discord_datamining.py:97]: line 91 and 92

File: cogs/discord_datamining.py (hunk 5 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py]: i made it more for like debugging
kae could easily remove it

File: cogs/discord_datamining.py (hunk 4 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py]: why dont you do it
all my variables in c# are just called bob
like bobthestringbuilder
bobtheembedbuilder
etc.

File: cogs/discord_datamining.py (hunk 3 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py:92]: i agree this can be kinda removed but its 3am so i was like speedrun

File: cogs/nickname_handler.py (hunk 2 above)
Comment (Silverdimond) [reply cogs/nickname_handler.py:210]: ok max you fix the typos

Comment (Silverdimond) [reply cogs/nickname_handler.py]: fixed

File: cogs/nickname_handler.py (hunk 1 above)
Comment (Silverdimond) [reply cogs/nickname_handler.py]: fixed

File: cogs/discord_datamining.py (hunk 5 above)
Comment (4Kaylum) [reply cogs/discord_datamining.py]: Log line is fine - that's the point of log lines.

File: cogs/discord_datamining.py [hunk 6]
```diff
@@ -74,18 +74,30 @@ async def commit_poster_loop(self, embed_title:str, api_url:str, repo_url:str):
...
-            embed.description = '\n'.join([f"{i['commit']['message']} - [Link]({repo_url}commit/{i['sha']})" for i in new_commits])
+            embed.description = array.pop()
             for sha, body in comment_text:
                 embed.add_field(sha, body, inline=False)
 
         # And send
         channel = self.bot.get_channel(self.VFL_CODING_CHANNEL_ID)
         m = await channel.send(embed=embed)
         await m.publish()
+        if len(array)>0:
```
Comment (4Kaylum) [inline cogs/discord_datamining.py]: This `if len(array) > 0` isn't necessary - Python is quite happy to iterate through an empty list without you explicitly checking it.

Comment (Silverdimond) [reply cogs/discord_datamining.py]: Didn't know that
Ill try to fix that tomorrow

File: cogs/discord_datamining.py (hunk 4 above)
Comment (fadedmax) [reply cogs/discord_datamining.py]: Yes but i’d like to keep these variable names somewhat descriptive, “pog” doesn’t give me anything.

File: cogs/discord_datamining.py (hunk 6 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py]: Ok fixed

File: cogs/discord_datamining.py (hunk 4 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py]: I think i made them more descriptive.

File: cogs/discord_datamining.py (hunk 3 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py:97]: Max I don't see why this is wrong.

File: cogs/nickname_handler.py [hunk 7]
```diff
@@ -128,7 +128,7 @@ async def on_member_update(self, before:discord.Member, member:discord.Member):
             if member.nick == new_nickname:
                 return
             try:
-                await member.edit(nick=new_nickname)
+                await member.edit(nick=new_nickname,reason= "Changed by Apple.Py automagically")
```
Comment (fadedmax) [inline cogs/nickname_handler.py:131]: "automatically"

File: cogs/nickname_handler.py [hunk 8]
```diff
@@ -94,7 +94,7 @@ async def on_member_join(self, member:discord.Member):
             )
         if data:
             try:
-                await member.edit(nick=data[0]["nickname"])
+                await member.edit(nick=data[0]["nickname"],reason= "Changed by Apple.Py automagically")
```
Comment (fadedmax) [inline cogs/nickname_handler.py:97]: "automatically"

**Review by fadedmax — CHANGES_REQUESTED** [review]
Just a few typos

**Comment by Silverdimond** [comment]
> Just a few typos

They aren't typos

File: cogs/discord_datamining.py [hunk 9]
```diff
@@ -75,17 +75,37 @@ async def commit_poster_loop(self, embed_title:str, api_url:str, repo_url:str):
             comment_text.extend([(i['commit_id'], re.sub(r"^## (.+)", r"**\1**", i['body'])) for i in data])
         self.logger.info(f"Logging info of {len(comment_text)} comments")
 
-        # Format it all into an embed
+        # Get the full description
+        descriptionunsplit =  '\n'.join([f"{i['commit']['message']} - [Link]({repo_url}commit/{i['sha']})" for i in new_commits])
+
+        # Inform the bot owner that the description is larger than allowed by discord
+        if len(descriptionunsplit)>2048:
```
Comment (fadedmax) [inline cogs/discord_datamining.py]: this is mainly a personal gripe, but there should be spaces:
`if len(descriptionunsplit) > 2048:`

File: cogs/nickname_handler.py (hunk 7 above)
Comment (Silverdimond) [reply cogs/nickname_handler.py:131]: Not a typo

File: cogs/nickname_handler.py (hunk 8 above)
Comment (Silverdimond) [reply cogs/nickname_handler.py:97]: Not a typo

File: cogs/discord_datamining.py (hunk 9 above)
Comment (Silverdimond) [reply cogs/discord_datamining.py]: fixed

**Review by fadedmax — APPROVED** [review]
LGTM

**Pull request merged**